mica-core/src/main/java/org/obiba/mica/user/UserProfileService.java

```diff
@@ -61,42 +61,54 @@ public synchronized Subject getProfileByApplication(@NotNull String username, @N
     return getProfileInternal(getProfileServiceUrlByApp(username, application, group));
   }
 
-  public synchronized List<Subject> getProfilesByApplication(@NotNull String application,
-    @Nullable String group) {
+  public synchronized List<Subject> getProfilesByApplication(@NotNull String application, @Nullable String group) {
+
     Assert.notNull(application, "Application name cannot be null");
+
     try {
-      RestTemplate template = newRestTemplate();
-      HttpHeaders headers = new HttpHeaders();
-      headers.set(APPLICATION_AUTH_HEADER, getApplicationAuth());
-      HttpEntity<String> entity = new HttpEntity<>(null, headers);
-      ResponseEntity<Subject[]> response =
-        template.exchange(getProfileServiceUrlByApp(null, application, group), HttpMethod.GET, entity, Subject[].class);
-
-      return Arrays.asList(response.getBody());
+
+      String profileServiceUrl = getProfileServiceUrlByApp(null, application, group);
+      return Arrays.asList(executeQuery(profileServiceUrl, Subject[].class));
+
     } catch(RestClientException e) {
       log.error("Agate connection failure: {}", e.getMessage());
     }
 
     return Lists.newArrayList();
   }
 
+  public String getUserManagementTranslations(String locale) {
+
+    String serviceUrl = UriComponentsBuilder
+      .fromHttpUrl(getAgateUrl())
+      .path(DEFAULT_REST_PREFIX)
+      .path("/config/i18n/" + locale + ".json")
```

Comment:
You should get the translations form a resource that is specific for the user related fields, instead of getting the whole agate thing.